### DISCUSSION TITLE
How to export implicit include directories to compile_commands.json?

### DISCUSSION THREAD

**Discussion by stevalkr** [mesonbuild/meson#13003] [Q&A]
I'm using `nix` as main package manager that the compiler is using non standard paths. With `cmake`, I use the following to help `clangd` find the standard headers, how can I do that with `meson`?

```cmake
if(CMAKE_EXPORT_COMPILE_COMMANDS)
  set(CMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES 
      ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES})
endif()
```

**Comment by ditsuke** [comment]
Bump

**Reply by ditsuke** [reply]
@etherswangel you might also want to share this on meson's matrix https://app.element.io/#/room/#mesonbuild:matrix.org
